src/tools/nbench/benchmark.cpp

```diff
@@ -214,7 +214,7 @@ vector<runtime::PerformanceCounter> run_benchmark(shared_ptr<Function> f,
     {
         for (int i = 0; i < warmup_iterations; i++)
         {
-            backend->call(f, results, args);
+            backend->call(backend->compile(f), results, args);
```

Comment:
Move compile outside the for loop. You don't want to call compile for every iteration.

Reply:
Maybe create a variable to hold backend->compile(f) outside the loop, and pass this var to backend->call in warm up and in the run.